python/ql/test/query-tests/Expressions/Regex/test.py

```diff
@@ -35,6 +35,9 @@
 re.compile(b"\$  ")
 re.compile(b"abc$(?m)")
 re.compile(b"abc$()")
+re.compile(b"((a$)|b)*")
+re.compile(b"((a$)|b){4}")
+re.compile(b"((a$).*)")
```

Comment:
Would have been cool to see the actual FPs results in the .expected file, that would be given _before_ the problem was fixed. I guess they would just be _Regular expressions containing a dollar '$' in the middle cannot be matched, whatever the input._

- nitpick: Can we remove the `Likely` part in line 28? All of these are FPs for unmatchable dollar the way I see it. 
- nitpick: Can we add a link to the FP report? That way, in 2 years time when I question whether anyone would ever write a regex such as `"((a$)|b){4}"`, we can easily answer that question :wink:

Reply:
It seems that would require a new test referring to `UnmatchableDollar.ql`, is that what you meant?
np1: Being tested, they should not be Fps any more, right? It should just be `OK uses of $ in the middle` or something?
np2: You mean inside `test.py`?

Reply:
> np1: Being tested, they should not be Fps any more, right? It should just be OK uses of $ in the middle or something?

What I'm trying to suggest is that you add your new cases to the python file _first_, then run queries and update .expected files and commit that. That will clearly show that the existing query reported alerts on the new cases. Then you fix the ql code, update the expected files again, and make a second commit. 

> np2: You mean inside test.py?

yep :+1: